lib/js/components/DatePickers/DatePicker/DatePicker.composables.ts

```diff
@@ -0,0 +1,108 @@
+import { onUnmounted, Ref, ref, watch } from 'vue';
+import { FlatpickrFn, Instance as DatePickerInstance } from 'flatpickr/dist/types/instance';
+import { CustomLocale } from 'flatpickr/dist/types/locale';
+
+import { DatePickerCalendarPositions, FLATPICKR_POSITIONS } from './index';
+
+let flatpickrFunction: FlatpickrFn | null = null;
+let locale: CustomLocale;
+
+export function initFlatpickr(
+	flatpickrInputRef: Ref<HTMLInputElement>,
+	dateRangePickerRef: Ref<HTMLElement>,
+	props: {
+		disableDates: Array<Date>;
+		minDate: Date | null;
+		maxDate: Date | null;
+		calendarPosition: DatePickerCalendarPositions;
+	},
+	onChange: (dates: Array<Date>) => void,
+	defaultDates: Date | Array<Date>,
+	mode: 'single' | 'range' = 'single',
+): {
+	datePicker: DatePickerInstance | null;
+	createDatePicker: () => Promise<DatePickerInstance | undefined>;
+	isOpen: Ref<boolean>;
+	toggle: () => void;
+} {
```

Comment:
```suggestion
interface InitFlatpickrPrams {
	flatpickrInputRef: Ref<HTMLInputElement>;
	dateRangePickerRef: Ref<HTMLElement>;
	props: {
		disableDates: Array<Date>;
		minDate: Date | null;
		maxDate: Date | null;
		calendarPosition: DatePickerCalendarPositions;
	};
	onChange: (dates: Array<Date>) => void;
	defaultDates: Date | Array<Date>;
	mode: 'single' | 'range' = 'single';
}

interface InitFlatpickr {
	datePicker: DatePickerInstance | null;
	createDatePicker: () => Promise<DatePickerInstance | undefined>;
	isOpen: Ref<boolean>;
	toggle: () => void;
}

export function initFlatpickr({
	flatpickrInputRef,
	dateRangePickerRef,
	props,
	onChange,
	defaultDates,
	mode = 'single',
}: InitFlatpickrPrams): InitFlatpickr {
```

Optional: in my opinion, using destructors and separate interfaces are more readable :)